.github/workflows/ci.yml

```diff
@@ -0,0 +1,71 @@
+name: CI
+
+on:
+  push:
+    branches: [master, staging]
+  pull_request:
+    branches: [master, staging]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CHROME_BIN: /usr/bin/google-chrome
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  test:
+    name: Checks & Build Test
+    runs-on: ubuntu-latest
+    env:
+      environment: ${{ github.ref == 'refs/heads/master' && 'prod' || 'stage' }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 12
+      - name: Test Variable
+        run: echo "Environment equals $environment."
+      - name: Install Dependenices
+        run: yarn
+      - name: Run Lint & Prettier Tests
+        run: |
+          yarn lint:type
+          yarn prettier:check
+      - name: Build Staging
+        run: yarn build:stage
+      - name: Build Production
+        run: yarn build:prod
+      - name: Run Test with CodeCov
+        run: yarn test:codecov
+      - name: Code coverage test
+        uses: codecov/codecov-action@v1.1.1
+
+  deploy:
+    name: Deployment
+    runs-on: ubuntu-latest
+    if: |
+      (github.ref == 'refs/heads/staging' || github.ref == 'refs/heads/master')
+      && github.event_name == 'push'
+    env:
+      environment: ${{ github.ref == 'refs/heads/master' && 'prod' || 'stage' }}
```

Comment:
if you can't access this to directly set the `environment:`, you might want to define this globally to see if that works.

also, being an environment variable we should use all uppercase for the name

Reply:
I tried to set it globally, but it still failed. I got it working when I set the environment to
```environment: ${{ github.ref == 'refs/heads/master' && 'prod' || 'stage' }}```
It's not the best-looking code. I also found other people having the same issue on forums.